adapters/Vungle/VungleAdapter/GADMAdapterVungleInterstitial.m

```diff
@@ -25,34 +27,89 @@ - (instancetype)initWithGADMAdNetworkConnector:(id<GADMAdNetworkConnector>)conne
   self = [super init];
   if (self) {
     self.connector = connector;
+    self.adapterAdType = Unknown;
   }
   return self;
 }
 
 - (void)dealloc {
-  [self stopBeingDelegate];
+  //[self stopBeingDelegate];
 }
 
+#pragma mark - GAD Ad Network Protocol Banner Methods (MREC)
+
 - (void)getBannerWithSize:(GADAdSize)adSize {
-  NSError *error = [NSError
-      errorWithDomain:kGADMAdapterVungleErrorDomain
-                 code:0
-             userInfo:@{NSLocalizedDescriptionKey : @"Vungle doesn't support banner ads."}];
-  [_connector adapter:self didFailAd:error];
-}
+  self.adapterAdType = MREC;
 
-- (void)loadInterstitialAd {
-  NSError *error = [[VungleRouter sharedInstance] loadAd:desiredPlacement withDelegate:self];
-  if (error) {
-    [_connector adapter:self didFailAd:error];
+  // Check if given banner size is in MREC.
+  if (!CGSizeEqualToSize(adSize.size, kGADAdSizeMediumRectangle.size)) {
+    NSError *error = [NSError
+        errorWithDomain:kGADMAdapterVungleErrorDomain
+                   code:0
+               userInfo:@{
+                 NSLocalizedDescriptionKey : @"Vungle only supports banner ad size in 300 x 250."
+               }];
+    [self.connector adapter:self didFailAd:error];
+    return;
+  }
+
+  id<GADMAdNetworkConnector> strongConnector = self.connector;
+  self.desiredPlacement = [GADMAdapterVungleUtils findPlacement:[strongConnector credentials]
+                                                  networkExtras:[strongConnector networkExtras]];
+
+  if (!self.desiredPlacement) {
+    [strongConnector
+          adapter:self
+        didFailAd:[NSError errorWithDomain:kGADMAdapterVungleErrorDomain
+                                      code:0
+                                  userInfo:@{
+                                    NSLocalizedDescriptionKey : @"'placementID' not specified"
+                                  }]];
+    return;
+  }
+
+  // Check if a banner (MREC) ad has been initiated with the samne PlacementID
+  // or not. (Vungle supports only one banner currently.)
+  if (![[VungleRouter sharedInstance] canRequestBannerAdForPlacementID:self.desiredPlacement]) {
+    NSError *error =
+        [NSError errorWithDomain:@"google"
+                            code:0
+                        userInfo:@{
+                          NSLocalizedDescriptionKey : @"A banner ad type has been already "
+                                                      @"instanciated. Multiple banner ads are not "
```

Comment:
nit:
`instanciated` -> `instantiated`

Reply:
@joshuajplagonera, got it, I will change typo. 🙂

Reply:
@joshuajplagonera Thanks, done. 😊